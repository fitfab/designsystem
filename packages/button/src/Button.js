import styled from 'styled-components';

const ButtonView = styled.button`
  background: ${p => p.primary? 'blue': 'gray'};
  border: none;
  border-radius: 4px;
  color: #fff;
  cursor: pointer;
  font-size: 1rem;
  font-weight: 300;
  line-height: ${p => p.size === 'large'? '36px': '28px'};
  padding: ${p => p.size === 'large'? '0 32px': '0 16px'};
`;

const Button = ({ primary, backgroundColor, size, label, ...props }) => {
    console.log(primary, size, label);
    return (
        <ButtonView 
            primary={primary}
            size={size}
            type="button"
            {...props}
        >
            {label.toUpperCase()}
        </ButtonView>
    )
}


export default Button;