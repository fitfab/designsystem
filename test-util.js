import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

const themeDemo = {
  colors: {
    primary: 'blue',
    secondary: 'gray'
  }
}

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={themeDemo}>
        {children}
    </ThemeProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }