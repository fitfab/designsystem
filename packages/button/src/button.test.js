import React from 'react';
import { cleanup, render } from '../../../test-util'
import 'jest-styled-components';


import Button from './index';

describe('Button', () => {
    test('Render Correctly', () => {
        const { container } = render(<Button size="small" label="hello" />)
        expect(container).toMatchSnapshot()
    })
});