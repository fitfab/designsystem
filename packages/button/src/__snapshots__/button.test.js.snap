// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render Correctly 1`] = `
.c0 {
  border: none;
  border-radius: 4px;
  color: #fff;
  cursor: pointer;
  font-size: 1rem;
  font-weight: 300;
  line-height: 28px;
  padding: 0 16px;
}

<div>
  <button
    class="c0"
    type="button"
  >
    HELLO
  </button>
</div>
`;
